File: box/src/main/java/de/qabel/box/storage/ProgressListener.java
package de.qabel.box.storage;

import java.util.function.Consumer;

public abstract class ProgressListener implements Consumer<Long> {
    @Override
    public void accept(Long progress) {
        setProgress(progress);
    }

    @Override
    public Consumer<Long> andThen(Consumer<? super Long> after) {
        return this::setProgress;
    }

    public abstract void setProgress(long progress);

    public abstract void setSize(long size);
}

File: box/src/main/java/de/qabel/box/storage/ProgressListener.kt
package de.qabel.box.storage

import de.qabel.core.util.Consumer

abstract class ProgressListener : Consumer<Long> {
    override fun accept(progress: Long) {
        setProgress(progress)
    }

    abstract fun setProgress(progress: Long)

    abstract fun setSize(size: Long)
}
